src/panel/PanelManager.jsx

```diff
@@ -159,12 +168,14 @@ export default class PanelManager extends React.Component {
   }
 
   render() {
-    const { ActivePanel, options, isMinified } = this.state;
-
-    return <div className={classnames('panel_container',
-      { 'panel_container--hidden': isMinified }
-    )}>
-      <ActivePanel {...options} />
-    </div>;
+    const { ActivePanel, options, isMinified, isMobile } = this.state;
+
+    return <DeviceContext.Provider value={isMobile}>
```

Comment:
Would it be possible to move this provider higher in the component tree ? This way we would only need to implement this logic only once.

Reply:
I can move it one level higher, but this only wouldn't solve the fundamental problem, which is that non-React code (from the map) needs to know if we are in mobile or not, so we need to make it available outside of the context mechanism.
But I'll try something, it's true that the logic is duplicated.

Reply:
@G-Ray @amatissart In https://github.com/QwantResearch/erdapfel/pull/520/commits/1d0a6a8c11596507a187418341add31066eb11ba I've refactored the device test, so it's not duplicated and use the same shared media query.
I think we can't go further, as some non-React components will still need to access this information outside of the React Context.